content-elements/form/form-field/prototype/form-field.js

```diff
@@ -46,22 +46,101 @@ Alpine.data('formField', () => ({
       this.requiredErrorElement.style.display = "none";
     }
     this._validateMailInput();
+    this._validateNumberInput();
+    this._validateTextInput();
+    this._validateDateTimeInput();
+  },
+
+  _validateNumberInput() {
+    if (this.inputEl.type === 'number') {
+      let inputValue = parseInt(this.inputEl.value);
+      let minValue = parseInt(this.inputEl.min);
+      let maxValue = parseInt(this.inputEl.max);
+      let valid = minValue <= inputValue && inputValue <= maxValue && inputValue != '';
+      // don't check required validation
+      if (this.inputEl.value == '' || this.inputEl.value == this.inputEl.defaultValue) {
+        this.invalidErrorElement.setAttribute('hidden', 'true');
+        this.invalidErrorElement.setAttribute('aria-hidden', 'true');
+        return;
+      }
+      this.calculateVisibility(valid);
+      this.inputEl.setCustomValidity(valid ? '' : this.errorMessageInvalid);
+    }
+  },
+
+  _validateTextInput() {
+    if (this.inputEl.type === 'text' || this.inputEl.type === 'password') {
+      let inputValue = this.inputEl.value;
+      let maxCharacters = parseInt(this.inputEl.maxLength);
+      let valid = inputValue.length <= maxCharacters;
+      this.calculateVisibility(valid);
+      this.inputEl.setCustomValidity(valid ? '' : this.errorMessageInvalid);
+    }
+  },
+
+  _validateDateTimeInput() {
+    if (this.inputEl.classList.contains('flatpickr-input')) {
+      let valid = true;
+      switch (this.inputEl.value.length) {
+        case 5: // only time
+          let placeholderVal = new Date('2000-01-01T' + this.inputEl.value);
+          let placeholderMin = new Date('2000-01-01T' + this.inputEl.min);
+          let placeholderMax = new Date('2000-01-01T' + this.inputEl.max);
+          if ((this.inputEl.min != "" && this.inputEl.max != "")) {
+            valid = new Date(placeholderVal).getTime() >= new Date(placeholderMin).getTime() && new Date(placeholderVal).getTime() <= new Date(placeholderMax).getTime();
+          }
+          else {
```

Comment:
code smell: use a if, else if, else if clause directly instead of nesting if's inside the else clause